cmd/root.go

```diff
@@ -240,13 +241,22 @@ func startServer(opts *common.Options) error {
 		os.Stderr.WriteString(fmt.Sprintf("\n  ** Can't create db directory: %s\n\n", dbPath))
 		os.Exit(1)
 	}
-	syncFromHeight := opts.SyncFromHeight
-	if opts.Redownload {
-		syncFromHeight = 0
+	var cache *common.BlockCache
+	if opts.NoCache {
+		lengthsName, blocksName := common.DbFileNames(dbPath, chainName)
+		os.Remove((lengthsName))
+		os.Remove((blocksName))
+	} else {
+		syncFromHeight := opts.SyncFromHeight
+		if opts.Redownload {
+			syncFromHeight = 0
+		}
+		cache = common.NewBlockCache(dbPath, chainName, saplingHeight, syncFromHeight)
 	}
-	cache := common.NewBlockCache(dbPath, chainName, saplingHeight, syncFromHeight)
 	if !opts.Darkside {
-		go common.BlockIngestor(cache, 0 /*loop forever*/)
+		if !opts.NoCache {
+			go common.BlockIngestor(cache, 0 /*loop forever*/)
+		}
 	} else {
 		// Darkside wants to control starting the block ingestor.
 		common.DarksideInit(cache, int(opts.DarksideTimeout))
```

Comment:
Is it okay for `cache` to be `nil` here if/when Darkside calls `StartIngestor`?

Reply:
I thought about this; it's not possible for `cache` to be `nil` in Darkside mode. See the if/else just above. I didn't see any reason to ever run Darkside without the cache, since performance isn't a possible problem. But if  you have a suggestion, let me know, I'd be happy to implement it.

Reply:
> I didn't see any reason to ever run Darkside without the cache, since performance isn't a possible problem.

I agree, I was thinking of updating the condition above to `if opts.NoCache && !opts.Darkside` so the cache is always enabled with Darkside, but that doesn't seem necessary either.